### DISCUSSION TITLE
Enable adding songs to a playlist when viewing another playlist

### DISCUSSION THREAD

**Discussion by yllekz** [navidrome/navidrome#3071] [Ideas]
![image](https://github.com/navidrome/navidrome/assets/28446638/9304d3cd-b19d-44f7-a852-9d0cdb29bcd9)

As you can see in this screenshot/example, if I select a song in an existing playlist, there is no menu option to add it to another playlist (or manage it, or do anything else with it other than remove it from the current playlist it's in).

Can we loosen up the menu options in this context so we can do more with playlist items without needing to navigate all the way to the song's proper song/album/artist page?

**Comment by deluan** [comment]
IIRC, the reason for this was a technical one: Playlists are handled differently by the server, they are not the same as the other places where you see song lists, so it required some changes in the way things work.

I understand that the current UX is not great, but we are avoiding to do any further work on the current UI, so we can focus on the rewrite. 

Rest assured that in the new UI we are taking these feedback into consideration, or at least understanding the full scope, we can avoid tying ourselves with some dumb technical decisions.

**Comment by lowgos** [comment]
Hi, I'm interested in this feature as well. Is Navidrome v55 the big refactor the team was waiting for?
Should we create an issue for this along this discussion ?

I would love to get a favorite button on songs in the playlist view as well if possible, maybe it's a separate issue.

**Reply by metalheim** [reply]
Both features ("Favorite" and "add to playlist") were recently added (like 2 weeks ago) and will be included in the next release.

**Reply by lowgos** [reply]
That's wonderful! Thank you so much for the quick answer and the amazing project!
